File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/command/memento_solution/turn_right_command.cpp
// turn_right_command.cpp

#include "driver.h"
#include "turn_right_command.h"

turn_right_command::turn_right_command(driver * a_driver)
: simple_driver_command(a_driver)
{
}

void turn_right_command::execute()
{
	simple_driver_command::execute();
	get_driver()->turn_right();
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/command/solution/TurnRightCommand.m
% TurnRightCommand.m

classdef TurnRightCommand < SimpleDriverCommand
	methods
		% constructor:
		function theTurnRightCommand = TurnRightCommand(aDriver)
			theTurnRightCommand@SimpleDriverCommand(aDriver);
		end
	end

	methods
		function execute(theTurnRightCommand)
			theTurnRightCommand.Driver.turnRight;
		end

		function undo(theTurnRightCommand)
			theTurnRightCommand.Driver.undoTurnRight;
		end
	end
end
